### PR TITLE
chore: standardise forwardRefs across components

### PR DESCRIPTION
**Pull request by madhurisandbhor**
### What does it do?

Standardising forwardRefs across components, so added forward refs to interactive and actionable components.

### Why is it needed?

Currently, a lot of components don’t have forwardRef some do, and it applies to the input field of that component, some edge cases use imperativeHandle e.g. TextInput, realistically we should add forwardRef to all the components in some capacity, [Ref](https://strapi-inc.atlassian.net/browse/DX-1182).

### CODE REVIEW

File: .changeset/rotten-eagles-matter.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@strapi/design-system': minor
```
Comment (jhoward1994) [inline .changeset/rotten-eagles-matter.md:2]: Finally, some representation for rotten eagles! 🙏🏻🦅

**Comment by jhoward1994** [comment]
Looks good. It looks to me like some components still don't use forwardRef e.g.

components/Alert/Alert.tsx
components/Badge/Badge.tsx

Is that intentional or should all compos be set up with forward ref?

File: packages/design-system/src/components/SimpleMenu/SimpleMenu.tsx [hunk 2]
```diff
@@ -23,51 +24,57 @@ interface SimpleMenuProps
...
 }
 
-const SimpleMenu = ({ children, onOpen, onClose, popoverPlacement, onReachEnd, ...props }: SimpleMenuProps) => {
-  /**
-   * Used for the intersection observer
-   */
-  const contentRef = React.useRef<HTMLDivElement>(null);
+const SimpleMenu = React.forwardRef<HTMLButtonElement, SimpleMenuProps>(
+  ({ children, onOpen, onClose, popoverPlacement, onReachEnd, ...props }, forwardedRef) => {
+    const triggerRef = React.useRef<HTMLButtonElement>(null);
```
Comment (remidej) [inline packages/design-system/src/components/SimpleMenu/SimpleMenu.tsx:29]: Where is triggerRef used?

**Review by remidej — APPROVED** [review]

**Review by remidej — APPROVED** [review]

**Review by remidej — APPROVED** [review]

File: packages/design-system/src/components/SimpleMenu/SimpleMenu.tsx (hunk 2 above)
Comment (madhurisandbhor) [reply packages/design-system/src/components/SimpleMenu/SimpleMenu.tsx:29]: Its passed as a composedRef to MenuTrigger

**Pull request merged**